### ISSUE TITLE
[Java] Implement new Concept Exercise: arrays

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1783]
This issue describes how to implement the `arrays` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of exercise is to learn about basic operations on arrays

## Learning objectives

- Defining an array.
- Accessing elements in an array by index.
- Updating an element in an array by index.
- Iterating over elements in an array.
- Basic array functions (like finding the index of an element in an array).

## Out of scope

- Multi-dimensional arrays.
- Memory and performance characteristics of arrays.
- Iterators.

## Concepts

- `arrays`: know how to define an array; know how to access elements in an array by index; know how to update an element in an array by index; know how to iterate over elements in an array; know of some basic functions (like finding the index of an element in an array).
- `for-loops`: know how to use a `for` loop to do iteration.

## Prerequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to work with fields.
- `booleans`: know what a `boolean` is.
- `basics`: know how to work with integer numbers and how to assign and update variables.

### Hints

Nothing to report

### After

Nothing to report

## Representer

This exercise does not require any specific representation logic to be added to the [representer](https://github.com/exercism/java-representer)

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer](https://github.com/exercism/java-analyzer)

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by samuelteixeiras** [comment]
Hi, I will start the implementation of this Concept.

**Comment by mirkoperillo** [comment]
Hi @samuelteixeiras good
Thank you for your help

I suggest to start from the C# version and take it as a blueprint
https://github.com/exercism/v3/tree/master/languages/csharp/exercises/concept/arrays

Feel free to ask for every doubt.

**Comment by mirkoperillo** [comment]
fix by PR #2075
